src/Datastore/Transaction.php

```diff
@@ -0,0 +1,455 @@
+ * never partially applied. Either all of the operations in the transaction are
+ * applied, or none of them are applied.
+ *
+ * It is highly recommended that users read and understand the underlying
+ * concepts in [Transactions](https://cloud.google.com/datastore/docs/concepts/transactions)
+ * before beginning.
+ *
+ * Mutations (i.e. insert, update and delete) are not executed immediately.
+ * Calls to those methods (and their batch equivalents) will enqueue a new
+ * mutation. Calling {@see Transaction::commit()} will execute all the mutations
+ * in the order they were enqueued, and end the transaction.
+ *
+ * Lookups and queries can be run in a transaction, so long as they are run
+ * prior to committing or rolling back the transaction.
+ *
+ * Transactions are an **optional** feature of Google Cloud Datastore. Queries,
+ * lookups and mutations can be executed outside of a Transaction from
+ * {@see Google\Cloud\Datastore\DatastoreClient}.
+ *
+ * @see https://cloud.google.com/datastore/docs/concepts/transactions Transactions
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder;
+ *
+ * $datastore = $cloud->datastore();
+ *
+ * $transaction = $datastore->transaction();
+ * ```
+ */
+class Transaction
+{
+    /**
+     * @var Operation
+     */
+    private $operation;
+
+    /**
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * @var string
+     */
+    private $transactionId;
+
+    /**
+     * @var array
+     */
+    private $mutations = [];
+
+    /**
+     * Create an Transaction
+     *
+     * @param Operation $operation Class that handles shared API interaction.
+     * @param string $projectId The Google Cloud Platform project ID.
+     * @param string $transactionId The transaction to run mutations in.
+     */
+    public function __construct(
+        Operation $operation,
+        $projectId,
+        $transactionId
+    ) {
+        $this->operation = $operation;
+        $this->projectId = $projectId;
+        $this->transactionId = $transactionId;
+    }
+
+    /**
+     * Insert an entity
+     *
+     * No service requests are run when this method is called.
+     * Use {@see Google\Cloud\Datastore\Transaction::commit()} to commit changes.
+     *
+     * Example:
+     * ```
+     * $key = $datastore->key('Person', 'Bob');
+     * $entity = $datastore->entity($key, ['firstName' => 'Bob']);
+     *
+     * $operation->insert($entity);
+     * ```
+     *
+     * @param Entity $entity The entity to insert.
+     * @return Transaction
+     */
+    public function insert(Entity $entity)
+    {
+        return $this->insertBatch([$entity]);
+    }
+
+    /**
+     * Insert multiple entities
+     *
+     * No service requests are run when this method is called.
+     * Use {@see Google\Cloud\Datastore\Transaction::commit()} to commit changes.
+     *
+     * Example:
+     * ```
+     * $keys = $datastore->keys(['kind' => 'person'], [
```

Comment:
It would be nice to use the shorthand for this example and any other similar use cases:

```php
$entities = [
    $datastore->entity('Person', ['firstName' => 'Bob']),
    $datastore->entity('Person', ['firstName' => 'John'])
];
```